client/shared/src/components/VirtualList.tsx

```diff
@@ -1,9 +1,16 @@
 import * as React from 'react'
 import VisibilitySensor from 'react-visibility-sensor'
 
-interface Props {
+interface Props<T, U> {
     itemsToShow: number
-    items: JSX.Element[]
+    items: T[]
+
+    // Additional props passed to the render function.
+    itemProps: U
+    // Function to render an item once it becomes visible.
```

Comment:
I think you've written too much Go 😄 
```suggestion
    /** Function to render an item once it becomes visible. */
```

Reply:
Argh, no I used <kbd>cmd</kbd> + <kbd>/</kbd>
But maybe it's also true that I do 😬